edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookService.java

```diff
@@ -781,4 +782,11 @@ public void finalizeGrades(String gradebookUid)
 	 */
 	void updateGradebookSettings(String gradebookUid, GradebookInformation gbInfo);
 
+	/**
+	 * Return the gradebook but with the GradeMappings. The normal getGradebook(siteId)
+	 * doesn't return the GradeMapping.
+	 * @param gradebookId
+	 * @return a gradebook with all the GradeMappings.
+	 */
+	public Set getGradebookGradeMappings(Long gradebookId);
```

Comment:
Ideally this shouldn't pass around the persistent objects either, but there isn't a clear separation of this in GradebookService and this one is a bit more complex to map out. Will leave this for now.